Generators/include/Generators/TPCLoopers.h

```diff
@@ -0,0 +1,146 @@
+// Static Ort::Env instance for multiple onnx model loading
+extern Ort::Env global_env;
+
+// This class is responsible for loading the scaler parameters from a JSON file
+// and applying the inverse transformation to the generated data.
+struct Scaler {
+  std::vector<double> normal_min;
+  std::vector<double> normal_max;
+  std::vector<double> outlier_center;
+  std::vector<double> outlier_scale;
+
+  void load(const std::string& filename);
+
+  std::vector<double> inverse_transform(const std::vector<double>& input);
+
+ private:
+  std::vector<double> jsonArrayToVector(const rapidjson::Value& jsonArray);
+};
+
+// This class loads the ONNX model and generates samples using it.
+class ONNXGenerator
+{
+ public:
+  ONNXGenerator(Ort::Env& shared_env, const std::string& model_path);
+
+  std::vector<double> generate_sample();
+
+ private:
+  Ort::Env& env;
+  Ort::Session session;
+  TRandom3 rand_gen;
+};
+#endif // GENERATORS_WITH_TPCLOOPERS
+
+namespace o2
+{
+namespace eventgen
+{
+
+#ifdef GENERATORS_WITH_TPCLOOPERS
+class GenTPCLoopers
+{
+ public:
+  GenTPCLoopers(std::string model_pairs = "tpcloopmodel.onnx", std::string model_compton = "tpcloopmodelcompton.onnx",
+                std::string poisson = "poisson.csv", std::string gauss = "gauss.csv", std::string scaler_pair = "scaler_pair.json",
+                std::string scaler_compton = "scaler_compton.json");
+
+  Bool_t generateEvent();
+
+  Bool_t generateEvent(double& time_limit);
+
+  std::vector<TParticle> importParticles();
+
+  unsigned int PoissonPairs();
+
+  unsigned int GaussianElectrons();
+
+  void SetNLoopers(unsigned int& nsig_pair, unsigned int& nsig_compton);
+
+  void SetMultiplier(std::array<float, 2>& mult);
+
+  void setFlatGas(Bool_t& flat, const Int_t& number, const Int_t& nloopers_orbit);
+
+  void setFractionPairs(float& fractionPairs);
+
+  void SetRate(const std::string& rateFile, const bool& isPbPb, const int& intRate);
+
+  void SetAdjust(const float& adjust);
+
+  unsigned int getNLoopers() const { return (mNLoopersPairs + mNLoopersCompton); }
+
+ private:
+  std::unique_ptr<ONNXGenerator> mONNX_pair = nullptr;
+  std::unique_ptr<ONNXGenerator> mONNX_compton = nullptr;
+  std::unique_ptr<Scaler> mScaler_pair = nullptr;
+  std::unique_ptr<Scaler> mScaler_compton = nullptr;
+  double mPoisson[3] = {0.0, 0.0, 0.0};    // Mu, Min and Max of Poissonian
+  double mGauss[4] = {0.0, 0.0, 0.0, 0.0}; // Mean, Std, Min, Max
+  std::vector<std::vector<double>> mGenPairs;
+  std::vector<std::vector<double>> mGenElectrons;
+  unsigned int mNLoopersPairs = -1;
+  unsigned int mNLoopersCompton = -1;
+  std::array<float, 2> mMultiplier = {1., 1.};
+  bool mPoissonSet = false;
+  bool mGaussSet = false;
+  // Random number generator
+  TRandom3 mRandGen;
+  // Masses of the electrons and positrons
+  TDatabasePDG* mPDG = TDatabasePDG::Instance();
+  double mMass_e = mPDG->GetParticle(11)->Mass();
+  double mMass_p = mPDG->GetParticle(-11)->Mass();
```

Comment:
same here. These are rather constants. Probably no need to have them as class member variables.